### PR TITLE
Installable

### PR DESCRIPTION
**Pull request by heavengate**
**make ppdet installable**

- add `setup.py` to package ppdet
- add `ppdet.model_zoo` module and `list_model` `get_config_file` `get_weights_url` `get_model` API
- add `ppdet.engine` module and `Detector` class which implements `train` `evaluate` `predict` `export` API

**TODO:** ppdet.metrics

**Examples**

1. 
```
>>> import ppdet
>>> ppdet.model_zoo.list_model('rcnn')
[12/29 15:12:04] ppdet.model_zoo.model_zoo INFO: Available Models:
        cascade_mask_rcnn_r50_fpn_1x_coco
        mask_rcnn_r50_fpn_1x_coco
        cascade_rcnn_r50_fpn_1x_coco
        faster_rcnn_r50_fpn_1x_coco
        faster_rcnn_r50_1x_coco
        mask_rcnn_r50_1x_coco

>>> ppdet.model_zoo.get_model('mask_rcnn_r50_1x_coco')
[12/29 15:12:18] ppdet.utils.download INFO: Downloading mask_rcnn_r50_1x_coco.yml from https://paddlemodels.bj.bcebos.com/object_detection/dygraph/configs/mask_rcnn_r50_1x_coco.yml
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 741.44KB/s]
[12/29 15:12:20] ppdet.utils.download INFO: Downloading mask_rcnn_r50.yml from https://paddlemodels.bj.bcebos.com/object_detection/dygraph/configs/./_base_/models/mask_rcnn_r50.yml
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 2130.17KB/s]
[12/29 15:12:23] ppdet.utils.download INFO: Downloading rcnn_1x.yml from https://paddlemodels.bj.bcebos.com/object_detection/dygraph/configs/./_base_/optimizers/rcnn_1x.yml
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 625.08KB/s]
[12/29 15:12:26] ppdet.utils.download INFO: Downloading coco_instance.yml from https://paddlemodels.bj.bcebos.com/object_detection/dygraph/configs/./_base_/datasets/coco_instance.yml
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 622.58KB/s]
[12/29 15:12:28] ppdet.utils.download INFO: Downloading mask_reader.yml from https://paddlemodels.bj.bcebos.com/object_detection/dygraph/configs/./_base_/readers/mask_reader.yml
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00, 1082.54KB/s]
[12/29 15:12:31] ppdet.utils.download INFO: Downloading runtime.yml from https://paddlemodels.bj.bcebos.com/object_detection/dygraph/configs/./_base_/runtime.yml
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 421.41KB/s]
W1229 15:12:34.258695  1507 device_context.cc:320] Please NOTE: device: 0, GPU Compute Capability: 6.1, Driver API Version: 10.1, Runtime API Version: 9.0
W1229 15:12:34.264214  1507 device_context.cc:330] device: 0, cuDNN Version: 7.6.
[12/29 15:12:39] ppdet.utils.download INFO: Downloading mask_rcnn_r50_1x_coco.pdparams from https://paddlemodels.bj.bcebos.com/object_detection/dygraph/mask_rcnn_r50_1x_coco.pdparams
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 300852/300852 [01:00<00:00, 4935.77KB/s]
<ppdet.modeling.architectures.mask_rcnn.MaskRCNN object at 0x7f0a854758e0>
```

2. 
```
import ppdet

cfg = ppdet.model_zoo.get_config_file('ssd_vgg16_300_240e_voc')
cfg = ppdet.core.workspace.load_config(cfg)
detector = ppdet.engine.Detector(cfg)
detector.train()
```

### CODE REVIEW

File: dygraph/ppdet/engine/evaluate.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+from __future__ import print_function
+
+import os
+import time
+import datetime
+
+import paddle
+from paddle.distributed import ParallelEnv
+
+from ppdet.utils.eval_utils import get_infer_results, eval_results
```
Comment (qingqing01) [inline dygraph/ppdet/engine/evaluate.py:26]: ppdet/utils/eval_utils.py   不需要放到utils里吧

**Comment by qingqing01** [comment]
提到master分支吧

File: dygraph/ppdet/core/workspace.py [hunk 2]
```diff
@@ -126,6 +126,7 @@ def load_config(file_path):
 
     # load config from file and merge into global config
     cfg = _load_config_with_base(file_path)
+    cfg['filename'] = os.path.splitext(os.path.split(file_path)[1])[0]
```
Comment (qingqing01) [inline dygraph/ppdet/core/workspace.py:129]: [1] -> [-1] ?

**Comment by heavengate** [comment]
@qingqing01  以上意见修改了，提到master PR: https://github.com/PaddlePaddle/PaddleDetection/pull/2034   当前PR
关闭

**Pull request closed without merging**